readme.md

```diff
@@ -0,0 +1,243 @@
+# remark-breaks
+
+[![Build][build-badge]][build]
+[![Coverage][coverage-badge]][coverage]
+[![Downloads][downloads-badge]][downloads]
+[![Size][size-badge]][size]
+[![Sponsors][sponsors-badge]][collective]
+[![Backers][backers-badge]][collective]
+[![Chat][chat-badge]][chat]
+
+[**mdast**][mdast] utility to support hard breaks without needing spaces or
+escapes (turns enters into `<br>`s).
+
+## Contents
+
+*   [What is this?](#what-is-this)
+*   [When should I use this?](#when-should-i-use-this)
+*   [Install](#install)
+*   [Use](#use)
+*   [API](#api)
+    *   [`newlineToBreak(tree)`](#newlinetobreaktree)
+*   [Syntax](#syntax)
+*   [Syntax tree](#syntax-tree)
+*   [Types](#types)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Related](#related)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## What is this?
+
+This package is a utility that takes an [mdast][] tree and turns soft line
+endings (enters) into hard breaks (`<br>`s)
+
+This package is used inside [remark-breaks][], which focusses on making it
+easier to transform content by abstracting these internals away.
+
+## When should I use this?
+
+This plugin is useful if you want to display user content closer to how it was
+authored, because when a user includes a line ending, it’ll show as such.
+GitHub does this in a few places (comments, issues, PRs, and releases), but it’s
+not semantic according to HTML and not compliant to markdown.
+Markdown already has two ways to include hard breaks, namely trailing spaces and
+escapes (note that `␠` represents a normal space):
+
+```markdown
+lorem␠␠
+ipsum
+
+lorem\
+ipsum
+```
+
+Both will turn into `<br>`s.
+If you control who authors content or can document how markdown works, it’s
+recommended to use escapes instead.
+
+## Install
+
+This package is [ESM only](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c).
+In Node.js (version 14.14+, or 16.0+), install with [npm][]:
```

Comment:
```suggestion
This package is [ESM only][esm].
In Node.js (version 16+), install with [npm][]:
```

Reply:
I’m getting conflicting messages. One comment says we need to support the same versions as `remark-breaks`, which is 12+. However, here you’re asking me to drop support for 16+. Either is fine with me.

Reply:
There is a difference between where code works, and what people should do.
Code should preferably work in Node 12, so we don’t need a major to `remark-breaks`.
People should not use Node 12, they should use Node 16 or later

Reply:
Indeed, that's what I suggest.

I think we've discussed this somewhere before?

This has to do with how these sections in the readme are worded. They are like "use maintained node and stuff works". They're not about "this current major release line is still compatible with Node 12, which you shouldn't use".

I'm open to figuring out how to solve/implement/test/document the latter.
Of interest perhaps: https://github.com/wooorm/import-meta-resolve/pull/15#issuecomment-1522157446